### ISSUE TITLE
Convert Starter Packs into Lists

### ISSUE THREAD

**Issue by PauloCarrasco** [bluesky-social/social-app#5970]
### Describe the Feature

A nice feature to have is a way to easily convert a starter pack into a list using a button like the "Follow All" on the top of the pack page (see image).

I think lots of people don't want to follow every account in a starter pack, and cherry picking accounts to follow or to manually add to a list is tedious. 

In my case, I like to add accounts associated to a subject to a list and scan that list for a while and then decide to cherry pick some accounts that create interesting content.

Hope my 2 cents of an idea can make the starter pack feature even more functional.

Kudos!

### Attachments

![Screenshot 2024-10-28 at 10 24 36](https://github.com/user-attachments/assets/4f7fdb8d-dab2-4423-b321-7d9617bf1cf3)

**Comment by yoavg** [comment]
an even better version than "convert to list" would be to have "add to list" option from a starter pack. this covers the "convert to list" use case, but also allows more flexibility in list management.

**Comment by svivian** [comment]
Honestly why are these two separate things in the first place? At the core they are identical - a collection of users. In the starter pack you can even look at the posts like a list.

Just make it so that you can either follow as a list, or follow all the accounts within.

**Comment by PauloCarrasco** [comment]
Same concept, different wording, right? The final objective is to have the option to follow some/all users or add them easily to a new/existing list.

**Comment by jbarciauskas** [comment]
Yeah I guess what this implies is that a Starter Pack should be a List plus a list of Feeds, where a List is a set of Accounts. Then you could just have an option to take a Starter Pack's List, or create a Starter Pack from a List (with an empty set of Feeds)

**Comment by yoavg** [comment]
I think that, if one takes the notion that users should be editing and
curating their own lists, then the semantics of "take a Starter
Pack's List" should be "add all the accounts in it to a list" rather than
"convert it to a user-list" (and similarly, when a user "takes" a list from
another user, they should have the ability to "add all accounts" to their
own list, rather than just duplicate the list).

**Comment by svivian** [comment]
> Starter Packs have both a group of `Accounts` and `Feeds`.

What’s the difference between a feed (within a starter pack) and a list? All I see is a list of recent posts by those users, which is exactly the same whether it’s a list or a feed in a starter pack.

**Comment by monicaellerose** [comment]
is there a known way to turn a list into a starter pack?

**Comment by monicaellerose** [comment]
> is there a known way to turn a list into a starter pack?

https://nws-bot.us/bskyStarterPack.php

**Comment by scottdotweb** [comment]
This is similar to #5940 but has had more discussion, so perhaps it should be merged in here.

**Comment by adamatic-me** [comment]
@joaomendoncaa Great work on getting this going! Please take a look at #5932, I tried to think a lot about reusing existing UI concepts there. 

I later added the part about doing the same for Lists, so that Lists can be combined... however, focusing on the just Starter Packs would be great at first!

**Comment by coastwalker404** [comment]
An random user here with a use case. I find lots of interesting starter packs that I want to add to a pinned list on my home page. I have to subscribe to the accounts in the starter pack and then go to the list of people I am subscribed to and individually move them to the pinned list. I would like for example to have a pinned feed on my home page with lists of related scientists from different starter packs combined. In my case this is different starter packs of microbiologists from slightly different fields. I could do the same with a feed on Astronomy and astrophysics or Public health and infectious disease doctors.

**Comment by coastwalker404** [comment]
Yes, your scenario would cover my case. I do take the point that others
have made that packaging an object as a "startup pack" and publishing it in
the way that is currently being used is also a useful tool. Thanks for your
response.

**Comment by OlofHarrysson** [comment]
I was also missing this feature so I decided to make a small web tool for it until there is an official solution - [pinkdawn.app](https://www.pinkdawn.app/).

Feel free to ping me on Bluesky [@oloflovesai.bsky.social](https://bsky.app/profile/oloflovesai.bsky.social) if you have any questions!

**Comment by cooljeanius** [comment]
Tonight I discovered that @sbm12 has created a tool for doing this: https://github.com/sbm12/bsky-Pack2List

**Comment by sbm12** [comment]
> An random user here with a use case. I find lots of interesting starter packs that I want to add to a pinned list on my home page. I have to subscribe to the accounts in the starter pack and then go to the list of people I am subscribed to and individually move them to the pinned list. I would like for example to have a pinned feed on my home page with lists of related scientists from different starter packs combined. In my case this is different starter packs of microbiologists from slightly different fields. I could do the same with a feed on Astronomy and astrophysics or Public health and infectious disease doctors.

While I agree this would be useful as an internally developed resource I've got a tool that combines lists, so you can do it now while waiting for the core dev to make it available: https://nws-bot.us/bskyListCombiner.php. One of several bits on my collection of tools here: https://github.com/sbm12/bsky-Pack2List.

**Comment by b0o** [comment]
Hey all, [I made a UserScript](https://gist.github.com/b0o/22521cf73f6c7a5aacbb8cd0f29197ac) for [TamperMonkey](https://www.tampermonkey.net/) which can do this directly from the starter pack page.

https://github.com/user-attachments/assets/858fbd65-d3b1-4ed8-bfd3-2673070242e6

**Comment by ed5464215** [comment]
I used @b0o script successfully.  Good job boo. What's the difference between using TamperMonkey and the other various and sundry methods of interacting with bluesky?

**Comment by Tamschi** [comment]
> I used @b0o script successfully. Good job boo. What's the difference between using TamperMonkey and the other various and sundry methods of interacting with bluesky?

Only lack of guardrails, essentially. Unlike OAuth or app passwords, TamperMonkey scripts and browser extensions can modify your access credentials. Browser extensions have some level of review (not much), while with TamperMonkey you really need to be sure you understand what it's doing and that it isn't able to run remote code in some manner.

**Comment by RufusIbiza** [comment]
It seems like this functionality will be native in the near future, but I'm working on a tool that will allow you to do it. If anyone would like to give it a test, please dm me on bsky @tarottherapy.space

Video of it in action [here](https://bsky.app/profile/tarottherapy.space/post/3lj5wlk6odk2m)

**Comment by NabilHunt** [comment]
I built [this ](https://pack2list.nabil.blue)some time ago
